Tech Doc changes to Operator README (#3)
MemVerge/memory-machine-operator: 0.0.8/README.md

# Memory Machine Operator

## Memory Machine Architecture
    
Before continuing, we recommend that you read [Memory Machine Container Architecture](architecture.md) to understand how Memory Machine™ Container is organized.

## Prerequisites

> The following instructions assume that you have installed the [OpenShift Container Platform](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html-single/installing/index#installation-process_ocp-installation-overview), customized to your requirements. 

Author: This is for the OpenShift marketplace, so we're only listing the OpenShift platform.

> The following instructions assume that you have installed the [OpenShift Container Platform](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html-single/installing/index#installation-process_ocp-installation-overview), customized to your requirements. 

Reviewer: What doc should Kubernetes users look into then?

> You must have a Kubernetes cluster deployed, and the `kubectl` command-line tool must be configured to communicate with your cluster.

Author: See above.

The [Operator SDK](https://sdk.operatorframework.io/docs/installation/) is a framework used to develop and run Operators. You will use the `operator-sdk` command to run the Memory Machine Operator, so you must have the Operator SDK installed.

## Configuring the Memory Machine Platform

Before installing Memory Machine Operator, complete the steps described below.

1. Obtain and install a [Memory Machine License](#obtaining-and-installing-a-memory-machine-license).
2. Configure [PMem](#configuring-pmem).

### Obtaining and Installing a Memory Machine License

> Obtain a Memory Machine license from MemVerge. To request a license, send an email to [mvlicense@memverge.com](mailto:mvlicense@memverge.com). Include the MAC address of the node that will contain the Memory Machine pod. This node is the *Memory Machine node*; the license is valid for the entire cluster but is installed only on the Memory Machine node.

Reviewer: > Include the MAC address of the node that will contain the Memory Machine pod. This node is the *Memory Machine node*; the license is valid for the entire cluster but is installed only on the Memory Machine node.

This is probably not very accurate. I think you mean the Memory Machine Management Center pod.

For example, a cluster has 10 worker nodes. We can have one Memory Machine pod on each worker node. That will be 10 MM pods in total.

There will be only one Memory Machine Management Center pod for the whole cluster. It will be on a particular worker node which the user choose to bind its MAC address to the license.

> Unless otherwise noted, type all commands on the Memory Machine node.

Reviewer: The commands will be on a client node, which is usually outside the K8S/OpenShift cluster, but has access to the cluster.

   ```
   $ kubectl label node <node-name> storage=pmem
   ```
    
    where `<node-name>` is the name of the worker node.

2. Run the [`config-pmem.sh`](scripts/config-pmem.sh) script to reserve PMem capacity (in GiB) for Memory Machine:
    
   ```
   $ config-pmem.sh --mm-capacity <pmem-size>
   ```
   where `pmem-size` is the number of GiB of PMem. The value of `pmem-size` cannot exceed the PMem capacity available on the node.
    
   If there is no physical PMem device, DRAM can be used to emulate PMem. In that case, add the `--pmem-emulation` flag as follows:
   
   ```
   $ config-pmem.sh --pmem-emulation --mm-capacity <pmem-size>
   ```

## Installing the Memory Machine Operator
    
Create an image-pull secret named `memverge-github-dockerconfig` and retrieve the Operator image by following these steps:

1. Log into Docker as follows:
    
   ```
   $ cat <file-containing-github-token> | docker login ghcr.io
   ```
   
2. Create the secret as follows:

   ```
   $ kubectl create secret generic memverge-github-dockerconfig \
       -n memverge \
       --from-file=.dockerconfigjson=$HOME/.docker/config.json \
       --type=kubernetes.io/dockerconfigjson
   ```
   

> 3. Update the pod credentials as follows:

Reviewer: ```suggestion
3. Patch the Service Account `default` as follows:
```

    
   ```
   $ kubectl patch serviceaccount default \
       -n memverge \
       -p '{"imagePullSecrets": [{"name": "memverge-github-dockerconfig"}]}'
   ```

> 4. Install the Memory Machine Operator node by typing the `operator-sdk` command as follows:

Reviewer: ```suggestion
4. Install the Memory Machine Operator by typing the `operator-sdk` command as follows:
```

    
   ```
   $ operator-sdk run bundle ghcr.io/memverge/memory-machine-operator-bundle:0.0.8 \
       -n memverge \
       --pull-secret-name memverge-github-dockerconfig
   ```

## Deployment Examples
    
The following examples are available:
    
- To deploy Memory Machine using a Memory Machine Operator that runs with a test pod, see [example.md](example.md).
    
- To deploy a Hazelcast cluster on Memory Machine and checkpoint-restore the Hazelcast cluster, see [tutorials/hazelcast/README.md](tutorials/hazelcast/README.md).

## Maintaining the Memory Machine Cluster

### Configuration Options
For a complete list of configuration options, refer to the [Memory Machine Configuration List](config.md).

> #### Changing Configurations on an Existing Memory Machine Container

Reviewer: We could only change all MM pods together.

```suggestion
#### Changing Configurations on Existing Memory Machine Pods
```

> To change the configuration of a running Memory Machine pod, redeploy the Memory Machine pod with the new configuration as follows:

Reviewer: ```suggestion
To change the configuration of running Memory Machine pods, redeploy the Memory Machine pods with the new configuration as follows:
```
